### ISSUE TITLE
A polygon causing crash

### ISSUE THREAD

**Issue by dmitriy-geels** [shapely/shapely#1364]
## Expected behavior and actual behavior.
Expected:
- `poly.is_valid` property return `True`/`False`
- `validation.explain_validity(poly)` return string
- `validation.make_valid(poly)` return new/same polygon

Actual:
- `poly.is_valid` read and `validation.make_valid(poly)` call causes `shapely.errors.PredicateError`
- `validation.explain_validity(poly)` call causes segfault

## Steps to reproduce the problem.
```
Python 3.9.7 (default, Sep 10 2021, 14:59:43)
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from shapely import wkt, validation
>>> t = 'MULTIPOLYGON (((13.98162568414809 49.66036193139579, 13.97835453878315 49.65783553058557, 13.97776670914051 49.6573815326369, 13.97573755346559 49.6558143566269, 13.973708152771 49.6562199329727, 13.9728355390253 49.6601000271289, 13.97450923745055 49.66325285192735, 13.97618293587584 49.6664056767258, 13.97336328676949 49.66824610672838, 13.9698080916865 49.6705666393369, 13.96980809168645 49.6705666393369, 13.96980809168678 49.67056663933709, 13.969808091687 49.6705666393372, 13.97039462387337 49.67090996381595, 13.97182124111153 49.67174502901957, 13.97238005237352 49.6720721271562, 13.97238005237352 49.67207212715621, 13.97141527697111 49.67423777787509, 13.969398 49.678766, 13.969068 49.679616, 13.969214 49.68015399999999, 13.97199799999999 49.679032, 13.97259663904885 49.67899768166424, 13.9746319999999 49.678881, 13.97498610533001 49.67823665979737, 13.97501933363488 49.67817619662194, 13.975386 49.677509, 13.97538600000007 49.677509, 13.9753860000001 49.677509, 13.977711 49.677771, 13.97797400000002 49.679316, 13.97852166719793 49.67962633363437, 13.979617 49.680247, 13.98152299767077 49.68221766425841, 13.98247600000002 49.683203, 13.9827941061485 49.686808705554, 13.98224700000003 49.688403, 13.98035400000004 49.690259, 13.98164984306604 49.69109284309143, 13.98335799999997 49.69219199999998, 13.98595799999998 49.690211, 13.98695299999997 49.689832, 13.98695299999998 49.68983200000001, 13.986953 49.689832, 13.988725 49.690612, 13.98872500000026 49.69061200000039, 13.98946955209382 49.69148582999485, 13.9900640531196 49.6921835552884, 13.99658203125 49.6936189256208, 13.99606704711914 49.7021685156335, 13.99657403163163 49.70229409094555, 13.99719123127126 49.70244696551158, 13.99719123127221 49.70244696550973, 13.99747962737293 49.7018863876096, 13.997735 49.70139, 13.99833358296499 49.70124212953971, 14.00060099999996 49.700682, 14.0006010000001 49.70068199999974, 14.00113551852914 49.69965370368327, 14.001257 49.69942, 14.00125700000001 49.69942000000001, 14.00125700000001 49.69942, 14.00251300000002 49.701765, 14.00256300000001 49.702449, 14.00147100000004 49.702913, 14.00154854415714 49.70331657553906, 14.0015908409596 49.7035367075965, 14.00205592687098 49.70365190501779, 14.00316428227888 49.70133260816175, 14.0053809999999 49.696694, 14.00538099999994 49.696694, 14.005381 49.696694, 14.008128 49.696771, 14.00972696617767 49.69830674510969, 14.01033562823308 49.69889134144955, 14.0117700000001 49.700269, 14.0133980000001 49.700457, 14.01374756243966 49.70180786226663, 14.01395979679023 49.70262802869009, 14.0158671187861 49.7022740510227, 14.015529 49.701033, 14.021649 49.698179, 14.0218657209063 49.6981439827543, 14.02186572090649 49.6981439827543, 14.02208643492191 49.69810832031161, 14.0247929999987 49.6976710000002, 14.024793 49.697671, 14.02479300000005 49.697671, 14.0247930000001 49.697671, 14.02479300000093 49.69767100000037, 14.02479300000304 49.69767100000126, 14.02712083327842 49.69865065785423, 14.02867137004718 49.69930319407229, 14.02867137004718 49.69930319407229, 14.03395915203066 49.6968804612029, 14.03427072108138 49.69590933688919, 14.034352 49.695656, 14.035797 49.695146, 14.03593999999998 49.694041, 14.03594 49.69404100000001, 14.03594 49.694041, 14.0368617823018 49.6945846042714, 14.03778356460361 49.69512820854279, 14.03809291204859 49.69498647308069, 14.04074277840002 49.69377236888182, 14.04074277840011 49.69377236888148, 14.04094483128803 49.69261328089078, 14.041085992922 49.6918034990737, 14.03946449213381 49.69084392287098, 14.03868899999998 49.690385, 14.03862235297361 49.689219647626, 14.038483 49.686783, 14.0396771224488 49.6856924141797, 14.039694 49.685677, 14.0396290505253 49.68560745548145, 14.03952371888488 49.6854946718408, 14.0350341796875 49.6837905907708, 14.02833938598633 49.6841237855088, 14.02833938598632 49.6841237855088, 14.0283393859863 49.6841237855088, 14.0275231425278 49.6838986861934, 14.02467718357944 49.68311384264665, 14.02310371398926 49.6826799195029, 14.02024126080425 49.68060425419788, 14.0181255288189 49.6790700626521, 14.0247344970703 49.6750711391359, 14.02386435351425 49.67458847979937, 14.02052879333496 49.6727382801951, 14.0167427368871 49.6722305841709, 14.01674273688707 49.6722305841709, 14.01100158691406 49.6714607174095, 14.00774002075195 49.6701275708694, 14.00640228169462 49.66778478131526, 14.0056065769961 49.666391259428, 14.00554414980003 49.66628193034614, 14.00541104296815 49.66604881964073, 14.00513761931507 49.6655699713371, 14.00513761931507 49.6655699713371, 14.00483462488558 49.66503933557491, 14.002263 49.66582, 14.00226299999998 49.66581999999999, 14.00226299999997 49.66582, 14.00130082042906 49.66560909140169, 13.9996169999999 49.66524, 13.9996169999991 49.6652399999977, 13.99932054523439 49.66439808245496, 13.99900509441485 49.66350221701298, 13.9987690783882 49.6628319426346, 13.998651562582 49.6628032359571, 13.99730284580899 49.6624737723834, 13.99656826139539 49.66229432862552, 13.98881260983371 49.66039978358037, 13.98702802621471 49.6599638467637, 13.98645213443338 49.6600735729912, 13.985248 49.660303, 13.983208 49.661584, 13.983208 49.661584, 13.98320799999999 49.661584, 13.98320799999988 49.66158399999991, 13.982902173301 49.6613659335929, 13.98278196186568 49.66125495834405, 13.98162568414809 49.66036193139579), (13.983208 49.661584, 13.985917 49.665267, 13.9833552649447 49.6617842129167, 13.983208 49.661584), (14.008194 49.695671, 14.0053612354975 49.6940361682191, 14.0025920000001 49.6924380000001, 14.0025920000001 49.692438, 14.008194 49.695671), (14.0025920000001 49.6924380000001, 14.002384 49.693939, 14.002592 49.692438, 14.0025920000001 49.6924380000001), (14.005686 49.67405, 14.004799 49.672564, 14.0047989999998 49.6725639999992, 14.0047990000001 49.672564, 14.005686 49.67405), (14.0041871844888 49.6705640869311, 14.0041871844887 49.670564086931, 14.003683 49.668916, 14.0041871844888 49.6705640869311), (13.999835 49.674254, 14.0035638179033 49.6690813310894, 14.0035638179033 49.6690813310895, 13.999835 49.674254), (13.983236 49.666989, 13.983121 49.676179, 13.983122 49.670407, 13.983236 49.666989), (14.018679 49.690227, 14.0143398812008 49.6819968235822, 14.019018 49.676032, 14.016214 49.679758, 14.0143398812009 49.6819968235822, 14.018679 49.690227)))'
>>> poly = wkt.loads(t)
>>> poly.is_valid
IllegalArgumentException: Cannot compute the quadrant for point (0,0)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/geometry/base.py", line 744, in is_valid
    return bool(self.impl['is_valid'](self))
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/predicates.py", line 25, in __call__
    return self.fn(this._geom)
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/geos.py", line 602, in errcheck_predicate
    raise PredicateError("Failed to evaluate %s" % repr(func))
shapely.errors.PredicateError: Failed to evaluate <_FuncPtr object at 0x7f0bd6779100>
>>> validation.explain_validity(poly)
IllegalArgumentException: Cannot compute the quadrant for point (0,0)

Segmentation fault (core dumped)
```
```
>>> validation.make_valid(poly)
IllegalArgumentException: Cannot compute the quadrant for point (0,0)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/validation.py", line 53, in make_valid
    if ob.is_valid:
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/geometry/base.py", line 744, in is_valid
    return bool(self.impl['is_valid'](self))
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/predicates.py", line 25, in __call__
    return self.fn(this._geom)
  File "/home/temporaryadmin/workspace/dh-ncr-hannibal/.venv/lib/python3.9/site-packages/shapely/geos.py", line 602, in errcheck_predicate
    raise PredicateError("Failed to evaluate %s" % repr(func))
shapely.errors.PredicateError: Failed to evaluate <_FuncPtr object at 0x7f14c16db4c0>
```
## Operating system
Ubuntu 21.10, kernel 5.17.1-051701-generic

## Shapely version and provenance
1.8.1post1 from pip

**Comment by mwtoews** [comment]
This is an upstream bug with GEOS 3.10.2 which can be reproduced with:
```bash
$ geosop -a 010600000001000000010300000009000000A2000000CBA844A497F62B40256F61BD86D44840FFEED7E2EAF42B40840165F433D4484088C08DD69DF42B40BDD5FB1325D448403F3A63DF93F32B40DBFA8EB9F1D34840020000E089F22B40EF8CC703FFD348401A00F17F17F22B40BD4D5E287ED448400100F1DFF2F22B40DC002E78E5D448400000F13FCEF32B40FAB3FDC74CD54840A49A37AC5CF22B40F309A31689D548401F61AAAF8AF02B40CCDEAC20D5D548400361AAAF8AF02B40CCDEAC20D5D54840BC61AAAF8AF02B40E7DEAC20D5D548403862AAAF8AF02B40F7DEAC20D5D548402D806B90D7F02B403593B060E0D54840A943C08D92F12B40DF71B9BDFBD548403E7458CCDBF12B4032589F7506D648403E7458CCDBF12B4034589F7506D64840F5D9DA575DF12B40C2D96A6C4DD64840999D45EF54F02B4081D1E5CDE1D6484044334FAE29F02B404FEB36A8FDD6484093A641D13CF02B402C1F49490FD74840038ECBB8A9F12B40FF774485EAD6484023D4C92FF8F12B4064676265E9D64840F0442BF702F32B4019709692E5D64840C4E4F86031F32B4095637875D0D64840EADBEDBB35F32B405DB5447ACED64840D8F335CB65F32B4072DF6A9DB8D64840FFF335CB65F32B4072DF6A9DB8D6484010F435CB65F32B4072DF6A9DB8D6484000E1438996F42B40AF963B33C1D6484041AD1402B9F42B4024D3A1D3F3D64840C315BECA00F52B40E13AE6FEFDD648401536035C90F52B40DF196D5512D748402289AE2E8AF62B40275F8EE852D74840C035221807F72B40B9C3263273D74840FE3701CA30F72B40347B0059E9D7484040302B14E9F62B40F1BBE9961DD848408301A1F5F0F52B4084622B685AD84840312FE8CE9AF62B40E7C9F3BA75D848400A8524B37AF72B40C04659BF99D848407F75AA7CCFF82B40732B84D558D84840BF6053E751F92B40FDDD3B6A4CD84840C56053E751F92B40FEDD3B6A4CD84840D06053E751F92B40FDDD3B6A4CD84840A779C7293AFA2B40D21C59F965D84840397AC7293AFA2B40091D59F965D84840EE42CDC09BFA2B40D7BC909B82D848401230F2ACE9FA2B407765827899D848400000000040FE2B40C8B64481C8D8484000000080FCFD2B40A0766DA8E0D948402BCD93F33EFE2B408164D4C5E4D9484009565CD98FFE2B40F9053CC8E9D948401F585CD98FFE2B40F5043CC8E9D948406BFC53A6B5FE2B403925C469D7D948402D26361FD7FE2B40E8DEC325C7D9484091AA529425FF2B40A879564DC2D9484080B036C64E002C40AC72A1F2AFD94840CFB036C64E002C408772A1F2AFD94840B0E4ADD594002C4026ECA7408ED948403BC8EBC1A4002C408C15359886D9484040C8EBC1A4002C408D15359886D9484040C8EBC1A4002C408C15359886D948403FA1496249012C401D3D7E6FD3D94840005C02F04F012C40D00D4DD9E9D9484001C991CEC0002C4075779D0DF9D94840CD0585F8CA002C4066560D4706DA48400CCAC383D0002C40CC54A77D0DDA4840005975790D012C40E6E8FF4311DA484063CEB1BF9E012C4028D65344C5D94840E809664CC1022C40E3A8DC442DD94840FE09664CC1022C40E3A8DC442DD94840200A664CC1022C40E3A8DC442DD94840205D6C5A29042C408EE6C8CA2FD94840AA3ED3EEFA042C4042698C1D62D94840487D22B64A052C40E07D7F4575D948401126AAB706062C4089B3226AA2D948401BAA471ADC062C408CA03193A8D94840F727A6EB09072C4063410CD7D4D948407F580DBD25072C400B3D1AB7EFD948403D8928BC1F082C4068FCB81DE4D948407847C66AF3072C4059350873BBD9484072FBE593150B2C4083E0F1ED5DD948405D64D8FB310B2C401AEA32C85CD94840C864D8FB310B2C401AEA32C85CD948408459C7E94E0B2C40E5660A9D5BD948400B5408ABB10C2C402A2F88484DD94840E75608ABB10C2C400E2F88484DD94840035708ABB10C2C400E2F88484DD948401F5708ABB10C2C400E2F88484DD94840F25808ABB10C2C40422F88484DD94840965D08ABB10C2C40BF2F88484DD9484032F127C8E20D2C400A687F626DD94840495E8903AE0E2C40B1435EC482D94840495E8903AE0E2C40B1435EC482D948403FBE0E1863112C403F0B036133D948407E8294EE8B112C40C675A18E13D94840B91AD99596112C4012A27C410BD94840FF9600FC53122C40FDF84B8BFAD848407F3C49BA66122C40A48AE255D6D848408A3C49BA66122C40A68AE255D6D848408A3C49BA66122C40A48AE255D6D84840C1EA2A8CDF122C404CD3F725E8D84840FD980C5E58132C40F31B0DF6F9D848408CEA06EA80132C40A0851651F5D84840728DC93CDC142C40A19B7188CDD84840A58DC93CDC142C40729B7188CDD8484008A58EB8F6142C40C0194F8DA7D84840ECD8273909152C40D4E35D048DD84840CC229EB034142C4084AFDB926DD84840026E6B0BCF132C40115322895ED848401BD51D4FC6132C403B28735938D8484068E7340BB4132C4071395E81E8D748404CA24E8F50142C403002DFC4C4D74840E2B19FC552142C40484F9143C4D7484094F447424A142C40E3972FFCC1D74840C00DF0733C142C40BA91164ABED7484000000000F0112C402856387386D7484001000080820E2C408896425E91D74840FBFFFF7F820E2C408896425E91D74840F0FFFF7F820E2C408896425E91D7484081146A83170E2C40307DFDFD89D74840B7DCE07CA20C2C405ECE3E4670D7484001000040D40B2C4050F33B0E62D748403D5E02105D0A2C4031A64A0A1ED74840F0FFD2BF47092C4048648FC4EBD64840F9FFFFFFA90C2C40F58728BB68D648408DC4D3F2370C2C40927F51EA58D64840FFFFFFBF820A2C40451AB8491CD6484052C0068192082C404731DBA60BD6484041C0068192082C404731DBA60BD64840FFFFFFFFA1052C4048E9BE6CF2D54840FEFFFF7FF6032C40188480BDC6D5484010FFEC2847032C40FE41C2F879D5484077AE81DDDE022C40B1CA0C4F4CD548408826CCAED6022C404F60EEB948D54840C12B793CC5022C40BBF2741641D5484003E0E565A1022C40D70B966531D5484003E0E565A1022C40D70B966531D54840438B17AF79022C40856C4A0220D5484050FBAD9D28012C40E882FA9639D5484045FBAD9D28012C40E782FA9639D5484040FBAD9D28012C40E882FA9639D54840CD664A80AA002C40CABEBFAD32D54840E70CA7CCCDFF2B40DAFE959526D54840250BA7CCCDFF2B4097FD959526D54840540148F1A6FF2B40A7D511FF0AD548402D1982987DFF2B40777201A4EDD44840E4DD1FA95EFF2B400C2E56ADD7D4484029C7F2414FFF2B40320F87BCD6D4484029EA857A9EFE2B40176CC9F0CBD448407DD9F5313EFE2B40EEE2800FC6D4484065377AA545FA2B40AF61E8FA87D44840BDA7C9BC5BF92B403C3201B279D44840E9EB104110F92B40BD79744A7DD4484019E6046D72F82B40B03907CF84D44840C554FA0967F72B4047C7D5C8AED44840C554FA0967F72B4047C7D5C8AED44840C054FA0967F72B4047C7D5C8AED448408254FA0967F72B403BC7D5C8AED4484080FD22F43EF72B4029BC8FA3A7D44840A69F82322FF72B40E933A200A4D44840CBA844A497F62B40256F61BD86D4484004000000C554FA0967F72B4047C7D5C8AED448401024EF1CCAF82B40D40D147827D54840E5925E577AF72B4040BE5758B5D44840C554FA0967F72B4047C7D5C8AED4484005000000FE0B040132042C4047E350BF0BD948407E4136B5BE022C40005F5A2DD6D848409C7616BD53012C400941F2CEA1D848409C7616BD53012C40FB40F2CEA1D84840FE0B040132042C4047E350BF0BD94840040000009C7616BD53012C400941F2CEA1D848403C11C47938012C40A0353FFED2D84840647616BD53012C40FB40F2CEA1D848409C7616BD53012C400941F2CEA1D848400500000012178046E9022C4035EF384547D64840651BB80375022C40C53BC09316D64840F41AB80375022C40543BC09316D648409D1BB80375022C40C53BC09316D6484012178046E9022C4035EF384547D648400400000028E298D224022C40DF9E430BD5D54840F0E198D224022C40D19E430BD5D54840331AF9BCE2012C40B5E21B0A9FD5484028E298D224022C40DF9E430BD5D5484004000000D5CA845FEAFF2B403D997FF44DD64840E9AFE21DD3012C4052B20175A4D54840E9AFE21DD3012C4060B20175A4D54840D5CA845FEAFF2B403D997FF44DD6484004000000ED7F80B56AF72B4058E542E55FD548408A05BEA25BF72B40F99E91088DD64840CBF44BC45BF72B40360186E5CFD54840ED7F80B56AF72B4058E542E55FD5484006000000753E3C4B90092C4079E8BB5B59D848406A92918E57072C4077A102AC4BD74840171230BABC092C405586713788D6484085798F334D082C40E198654F02D74840A292918E57072C4077A102AC4BD74840753E3C4B90092C4079E8BB5B59D84840 isValid -f txt
terminate called after throwing an instance of 'geos::util::IllegalArgumentException'
  what():  IllegalArgumentException: Cannot compute the quadrant for point (0,0)

Aborted (core dumped)
```
it is fixed for the next GEOS 3.10.x release and in the main branch. The geometry is reported as valid, but non-simple.

**Comment by jorisvandenbossche** [comment]
This is fixed in the GEOS version included in the latest wheels, so closing this issue.
